charms/kubernetes_snaps.py

```diff
@@ -607,6 +654,32 @@ def is_snap_installed(name):
     return call(cmd, stdout=DEVNULL, stderr=DEVNULL) == 0
 
 
+def is_upgrade(snap_name: str, target_channel: str):
+    """
+    Check if the installed version is less than the target channel version.
+
+    Args:
+    snap_name (str): Then name of the snap package.
+    target_channel (str): The target channel to compare against.
+
+    Returns:
+    bool: True if an upgrade is needed, False otherwise.
+    """
+    is_refresh = is_snap_installed(snap_name)
+
+    if is_refresh:
+        installed_version = get_snap_version(snap_name)
```

Comment:
Could this be changed to handle the chance the `get_snap_version(...)` result turns out to be None for some odd reason

```suggestion
    if is_refresh and (installed_version := get_snap_version(snap_name)):
```